src/main/java/bio/terra/cli/businessobject/User.java

```diff
@@ -192,6 +192,12 @@ public void fetchPetSaCredentials() {
       throw new SystemException("Error writing pet SA key to the global context directory.", ioEx);
     }
     logger.debug("Stored pet SA key file for this user and workspace.");
+
+    // Allow the pet to impersonate itself so that Nextflow and other app calls can run.
+    // TODO(PF-991): This will change in the future when WSM disallows SA self-impersonation
```

Comment:
nit: allow the pet and user to impersonate the pet right?
as an aside, I think we'll probably still have to call the WSM endpoint in the future, it's just the endpoint's behavior will have changed, so this particular call here may stay the same.

Reply:
Fixed comment.
I'm not sure, it depends on what changes. If we add identity/client SAs, I expect either WSM will return something different or the CLI will call a different service, so this spot would change. That depends a lot on future designs, though.

Reply:
Good point. Agreed we don't have a clear direction forward on this yet. TBDiscussed after you get back refreshed!